salt/states/file.py

```diff
@@ -929,10 +929,10 @@ def symlink(
                     return _error(ret, ((
                                             'File exists where the backup target {0} should go'
                                         ).format(backupname)))
-                elif os.path.isfile(backupname):
-                    os.remove(backupname)
-                elif os.path.isdir(backupname):
-                    shutil.rmtree(backupname)
+                elif os.path.isfile(backupname)\
```

Comment:
I don't understand the reasoning behind the original logic either. I left it there because I thought it was necessary for something I'm unaware of. To me it's saying to error out for anything that's not a file, a folder, or a symlink. Is there anything else that could be there? I can't think of anything in windows.